atproto/identity/nsid.go

```diff
@@ -0,0 +1,36 @@
+package identity
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+
+	"github.com/bluesky-social/indigo/atproto/syntax"
+)
+
+var (
+	ErrNSIDResolutionFailed = fmt.Errorf("NSID resolution mechanism failed")
+	ErrNSIDNotFound         = fmt.Errorf("NSID not associated with a DID")
+)
+
+// Resolves an NSID to a DID, as used for Lexicon resolution (using "_lexicon" DNS TXT record)
```

Comment:
no option for `https://*/.well-known/atproto-lexicon` ?

Reply:
nope. we discussed/debated and decided against. I think there is some discussion in: https://github.com/bluesky-social/atproto/discussions/3074